scripts/genmod/main.go

```diff
@@ -138,7 +139,7 @@ func updateModule(dir string, gomod *GoMod, modules map[string]*GoMod) error {
 			"-require", require.Path + "@" + *versionFlag,
 			"-replace", require.Path + "=" + relDir,
 		}
-		if *goVersionFlag != "" {
+		if *goVersionFlag != "" && semver.Compare("v"+*goVersionFlag, "v"+gomod.Go) == 1 {
```

Comment:
I am not sure if this would be acceptable for all cases since it might break the compatibility of agents if `go.mod` for a specific module is unintentionally upgraded. @axw can probably clarify this.

Alternatively, we can add another option to exclude certain modules and execute 2 separate commands in the update-modules (might be more of a hack as the list might grow).

Reply:
If `go.mod` is unintentionally upgraded, wouldn't PR reviews be there to prevent that?

Reply:
I think it's fine releasing new modules with a version greater than the agent's minimum, just not the other way around. If the version of a module is updated after release such that compatibility is broken, I would consider that a bug that we would fix by reverting.